### ISSUE TITLE
Can we bump to fairseq2 0.5?

### ISSUE THREAD

**Issue by jasonrichdarmawan** [facebookresearch/SONAR#78]
Goal: Rent 5090 because it has larger L2 cache

https://github.com/facebookresearch/SONAR/blob/8da24c2b1596561070e92e37ea79e57c426cd620/pyproject.toml#L27

**Comment by jasonrichdarmawan** [comment]
Wah the fairseq2 author just pretty much refactor the entire code base in 0.5, I can't even find fairseq2.data.text.tokenizers.TextTokenizer now

**Comment by avidale** [comment]
Bumping to fairseq2 0.5 is on the way (in a private branch); unfortunately, it also required refactoring most of the codebase. 
I hope we'll be able to merge it relatively soon. 

What does it have to do with 5090, though?

**Comment by jasonrichdarmawan** [comment]
> What does it have to do with 5090, though?

I am not sure about this. But, this is my hypothesis: to optimize GPU utilization, you need to match the batch size to the L2 cache. I got 30 minutes less in training time when I use the optimal batch size based on the L2 cache compared to random batch size that also show GPU utilization at 100%

Though, SAEs architecture is simple, so maybe that's why you can estimate the optimal batch size by matching the intermediate matrix shape with the L2 cache

<img width="1030" height="1106" alt="Image" src="https://github.com/user-attachments/assets/1d546ed2-cec1-4c2a-a6a0-761bfe3bee2b" />

5090 has L2 Cache of 96 MB. So it might be the fastest for SAEs training

Note: I don't have the privilege to use 4090 on a daily basis, let alone 5090 or A100. I use 3090 most of the time because it is more economical for me. So, my hypothesis could be wrong by a mile. For example, I did try to train 1 SAEs with A100 but I didn't take a look at the GPU utilization rate at the time and the training finished 1/3 than compared to a A4000*. Just to clarify, I do believe A100 should be faster than 5090, 4090, or A400. It's just that the moment you increase the batch size over the L2 cache, I assume it will only increase the training time

*I got access to A4000 from a fellowship. I rented A100 on my own out of curiosity with batch size 4096 and no gradient accumulation, while A4000 is set at batch size 128 and gradient accumulation of 32 steps.

**Comment by jasonrichdarmawan** [comment]
PS: I scaled the SAE to 131k, my mentor's assumption is because the SONAR vocab size is about 50k, then the SAE dimension should be larger than that. Though I doubt that the case

This time I was given A100 because 131k will take longer than the previous experiment. I accidentally set the batch size to 4096 and gradient accumulation 1 (i.e. without calculating the optimal batch size based on L2 cache). I run it side by side with a 4090. So far, A100 has not surpass the 4090 which was using optimal batch size based on L2 cache

PS 2: I am not sure why the metrics are not identical. Previously it was identical when I ran the experiment with and without accumulation gradient in the same machine. This time it's in a different machine

<img width="1792" height="1137" alt="Image" src="https://github.com/user-attachments/assets/86f8a084-7072-4371-a634-57bc3701534c" />
